src/value.hpp

```diff
@@ -380,7 +380,7 @@ class Value {
         } else if ((m_repr._bits & TAG_INT32) == TAG_INT32) {
             return (double)as_int32();
         } else {
-            return NAN;
+            return nan("");
```

Comment:
This is a sensitive change, but I verified that the binary pattern is the same (all zeroes at the end).